src/explore-education-statistics-common/src/components/form/rhf/FormProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  FieldValues,
+  FormProvider as RHFormProvider,
+  useForm,
+  UseFormProps,
+  UseFormReturn,
+} from 'react-hook-form';
+import React, { ReactNode, useEffect, useRef } from 'react';
+import { Schema } from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { isEqual } from 'lodash';
+
+interface FormProviderProps<TFormValues extends FieldValues> {
+  children: ReactNode | ((form: UseFormReturn<TFormValues>) => ReactNode);
+  initialValues?: UseFormProps<TFormValues>['defaultValues'];
+  validationSchema?: Schema<TFormValues>;
+}
+
+export default function FormProvider<TFormValues extends FieldValues>({
+  children,
+  initialValues,
+  validationSchema,
+}: FormProviderProps<TFormValues>) {
+  const form = useForm<TFormValues>({
+    defaultValues: initialValues,
+    resolver: validationSchema ? yupResolver(validationSchema) : undefined,
+  });
+
+  const previousInitialValues = useRef(initialValues);
+
+  /**
+   * RHF caches default values, to update them you you have to
+   * reset the form with the new values.
+   */
+  useEffect(() => {
+    if (!isEqual(previousInitialValues.current, initialValues)) {
+      form.reset(initialValues as TFormValues);
+      previousInitialValues.current = initialValues;
+    }
+  }, [form, initialValues]);
```

Comment:
It's needed because of the possibility of going back to a previous step in the table tool and changing the options so the filters/indicators are different. In the Formik version I think this is handled by the `enableReinitialize` prop, RHF doesn't have an equivalent but one of it's developers has recommended this approach in various github discussions.

I'll update the comment to make it clearer and maybe add an optional prop to enable it or not.

Reply:
Cool, yep that makes sense!